src/bp-core/classes/class-bp-email-recipient.php

```diff
@@ -90,7 +90,7 @@ public function __construct( $email_or_user, $name = '' ) {
 				$address = current( $email_or_user );
 			} else {
 				$address = key( $email_or_user );
-				$name    = current( $email_or_user );
+				$name    = wp_specialchars_decode( bp_core_get_user_displayname( $email_or_user ), ENT_QUOTES );
```

Comment:
@kent-carlo 
`bp_core_get_user_displayname` first argument only except user id or username as I know so if it's email we need to convert it in user_id or username.